snuba/datasets/events.py

```diff
@@ -200,6 +213,23 @@ def get_promoted_context_columns(self):
     def get_required_columns(self):
         return self.__required_columns
 
+    def column_expr(self, column_name, body):
+        if column_name in TIME_GROUP_COLUMNS:
+            return time_expr(column_name, body['granularity'])
+        elif NESTED_COL_EXPR_RE.match(column_name):
+            return  tag_expr(self, column_name)
```

Comment:
I think we should move the entire method into this class (same for tags_expr).
Both tag_expr and tags_expr take in the dataset in order to reference some events specific methods, like the list of promoted columns. Moving them here allows us to avoid passing the dataset through and remove those events specific methods from the dataset parent class since they will be only accessed from the events dataset itself.